### PR TITLE
[CGPROD-2928] Shop MVP loose ends

### PR DESCRIPTION
**Pull request by BeeGodwin**
- Scrollable list items now reflect the item's state in the collection on initial render
- Items update when their parent list becomes visible- ie if we now own or have equipped an item
- Scrollable list rebuilds itself if the underlying collection has changed- ie if we have different items
- Illegal transactions are now reflected in the UI - button is tinted & an alpha is applied
- Confirm button returns early (noop) on an illegal tx (we are still sending the click event, though.)

### CODE REVIEW

File: src/core/layout/scrollable-list/scrollable-list-buttons.js [hunk 1]
```diff
@@ -64,27 +65,50 @@ const scaleButton = (gelButton, layout, space) => {
     gelButton.setScale(scaleFactor);
 };
 
-const toggle = button => () => {
+const updateButton = button => {
```
Comment (FostUK) [inline src/core/layout/scrollable-list/scrollable-list-buttons.js:68]: Could one line this.

File: src/core/layout/scrollable-list/scrollable-list-buttons.js [hunk 2]
```diff
@@ -64,27 +65,50 @@ const scaleButton = (gelButton, layout, space) => {
...
+        return true;
+    };
+
+    return fp.isEqual(button.item, item) ? false : doUpdate(button, item);
+};
+
+const getState = (item, title) => {
+    const isShop = title => (title === "shop" ? true : false);
+    const isOwned = item => item.state && item.state === "owned";
+    const isEquipped = item => item.state && item.state === "equipped";
```
Comment (FostUK) [inline src/core/layout/scrollable-list/scrollable-list-buttons.js:87]: Can use optional chains now so this could be
item => item?.state === "equipped"

**Review by FostUK — APPROVED** [review]
Looks great. Some super minor comments but more just for future ref.

🎄🎄🎄🎄🎅🏿🤶🎄🎄🎄🎄

File: src/core/layout/scrollable-list/scrollable-list-buttons.js (hunk 2 above)
Comment (BeeGodwin) [reply src/core/layout/scrollable-list/scrollable-list-buttons.js]: 👍

File: src/core/layout/scrollable-list/scrollable-list-buttons.js (hunk 1 above)
Comment (BeeGodwin) [reply src/core/layout/scrollable-list/scrollable-list-buttons.js]: 👍

**Pull request merged**